Lib/test/test_syslog.py

```diff
@@ -78,6 +79,62 @@ def logger():
         finally:
             sys.setswitchinterval(orig_si)
 
+    def test_subinterpreter_syslog(self):
+        # syslog.syslog() is not allowed in subinterpreters, but only if
+        # syslog.openlog() hasn't been called in the main interpreter yet.
+        with self.subTest('before openlog()'):
+            code = dedent('''
+                import syslog
+                caught_error = False
+                try:
+                    syslog.syslog('foo')
+                except RuntimeError:
+                    caught_error = True
+                assert(caught_error)
+            ''')
+            res = support.run_in_subinterp(code)
+            self.assertEqual(res, 0)
+
+        syslog.openlog()
+        with self.subTest('after openlog()'):
+            code = dedent('''
+                import syslog
+                syslog.syslog('foo')
+            ''')
+            res = support.run_in_subinterp(code)
+            self.assertEqual(res, 0)
+        syslog.closelog()
```

Comment:
It may be helpful if we ensure `closelog()` gets called even if the test fails:

```suggestion
        syslog.openlog()
        try:
            with self.subTest('after openlog()'):
                code = dedent('''
                    import syslog
                    syslog.syslog('foo')
                ''')
                res = support.run_in_subinterp(code)
                self.assertEqual(res, 0)
        finally:
            syslog.closelog()
```